python-avd/pyavd/_cv/client/async_decorators.py

```diff
@@ -150,3 +104,238 @@ async def wrapper_cv_version(*args: Any, **kwargs: Any) -> list:
             raise LookupError(msg)
 
         return wrapper_cv_version
+
+
+class GRPCRequestHandler:
+    """
+    Decorator used to handle execution of the async gRPC calls towards CloudVision.
+
+    Retries an async method upon getting gRPC Status.UNAVAILABLE (14) using exponential backoff mechanism and max retry limit.
+    Converts GRPCError or AsyncioTimeoutError instances to an instance of the relevant subclass of CVClientException.
+    Splits gRPC messages into smaller chunks (based on reported maximum supported size) if Status.RESOURCE_EXHAUSTED is received.
+
+    Args:
+        max_retries (int): Maximum number of retry attempts for Status.UNAVAILABLE. Total attempts = 1 + max_retries.
+        initial_delay (int): Initial delay in seconds before the first retry.
+        factor (int): Multiplier for the delay in subsequent retries.
+        list_field (str): Name of the parameter to be split if Status.RESOURCE_EXHAUSTED is received.
+        min_items_for_splitting_attempt (int): Minimum length of the item that we'll still try to split.
+    """
+
+    max_retries: int
+    initial_delay: int
+    factor: int
+    list_field: str | None
+    min_items_for_splitting_attempt: int
+    func: Callable
+    func_signature: Signature
+    bound_arguments: BoundArguments
+    current_arguments_dict: dict
+
+    def __init__(
+        self,
+        max_retries: int = 5,
+        initial_delay: int = 1,
+        factor: int = 2,
+        list_field: str | None = None,
+        min_items_for_splitting_attempt: int = 2,
+    ) -> None:
+        self.max_retries = max_retries
+        self.initial_delay = initial_delay
+        self.factor = factor
+        self.list_field = list_field
+        self.min_items_for_splitting_attempt = max(2, min_items_for_splitting_attempt)
+
+    def __call__(self, func: Callable) -> Callable:
+        self.func = func
+        self.func_signature = signature(func)
+
+        def _is_list_annotation(annotation: Any, strict: bool = False) -> tuple[bool, Any]:
```

Comment:
move this to the class level as a @staticmethod, so we don't redefine it every time.